client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -220,7 +220,9 @@ public void resetReaderGroup(ReaderGroupConfig config) {
 
         return listOfMaps.stream()
                          .flatMap(map -> map.entrySet().stream())
-                         .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+                         .collect(Collectors.toMap(Entry::getKey,
+                                                   // A value of -1L implies read until the end of the segment.
+                                                   entry -> (entry.getValue() == -1L) ? Long.MAX_VALUE : entry.getValue()));
```

Comment:
I don't understand why we need this conversion from `-1` to `MAX_VALUE`.

Reply:
`getEndSegmentsForStreams(ReaderGroupConfig config)` returns the list of EndSegments and its corresponding end offsets upto which the segment should read. This `endOffset` is passed as parameter for `SegmentInputStreamImpl` and `Long.MAX_VALUE` corresponds to `SegmentInputStreamImpl#UNBOUNDED_END_OFFSET`

Reply:
The overloading of `-1` is confusing me. Could we use a different value to represent `UNBOUNDED`? Could we use `Long.MAX_VALUE` directly rather than convert some other value like `-1` to `Long.MAX_VALUE`?

Reply:
@fpj @tkaitchuck my reasons for not changing this is listed below. 
Please review it and let me know your opinion.

With PR https://github.com/pravega/pravega/pull/3364, `-1L` offset now represents a completely read segments and this can be part of a `StreamCut`. Now for bounded stream processing this StreamCut can used to represent the lower bound or the upper bound.

 In case of lower bound `-1L` implies that the we should start reading from the successors of the segments (with offset -1L) and in the case of upper bound the semantic changes to read until the end of the a segment.
Hence, changing -1L to Long.MAX_VALUE will transfer the confusion for lower bound.

Also, using `Long.MAX_VALUE` implies we will not be able to use `RevisionDataOutput#writeCompactLong` since it has a limit of `2^62-1`